crates/generated_parser/src/early_error_checker.rs

```diff
@@ -0,0 +1,646 @@
+use crate::context_stack::{
+    BindingInfo, BindingKind, BindingsIndex, BreakOrContinueIndex, ContextMetadata, LabelIndex,
+    LabelInfo, LabelKind,
+};
+use crate::declaration_kind::DeclarationKind;
+use crate::early_errors::*;
+use crate::error::{ParseError, Result};
+use crate::Token;
+use ast::arena;
+use ast::source_atom_set::{SourceAtomSet, SourceAtomSetIndex};
+use std::cell::RefCell;
+use std::rc::Rc;
+
+// Internal Helper Methods for Early Error Checker Labels/Controls
+
+/// Internal Label/Control Helper method:
+/// called From EarlyErrorChecker::check_labelled_statement, and not used by
+/// the struct implementing the trait. This means that
+/// LabelledStatementEarlyErrorsContext is allocated inside this trait.
+fn check_unhandled_continue<'alloc>(
+    context_metadata_mut: &mut ContextMetadata,
+    context: LabelledStatementEarlyErrorsContext,
+    index: BreakOrContinueIndex,
+) -> Result<'alloc, ()> {
+    for info in context_metadata_mut.breaks_and_continues_from(index) {
+        context.check_labelled_continue_to_non_loop(info)?;
+    }
+
+    Ok(())
+}
+
+/// Internal Label/Control Helper method:
+/// called From EarlyErrorChecker::check_script_bindings
+/// EarlyErrorChecker::check_module_bindings,
+/// EarlyErrorChecker::check_function_bindings, and not used by
+/// the struct implementing the trait. This means that
+/// the contexts associated with those methods are allocated inside this trait.
+fn check_unhandled_break_or_continue<'alloc, T>(
```

Comment:
(in contrast to above, this is surely checking "unhandled", at script/module/function level.)